### PR TITLE
Address some `&mfs` crashes

### PR DESCRIPTION
**Pull request by povik**
Yosys calls `&mfs` as part of the default abc9 flow but every time it does it's prepared for the worst, and is ready to recover when the command crashes (see YosysHQ/yosys#1974). In this PR we try to make it crash less, see the individual commits for details.

### CODE REVIEW

File: src/aig/gia/giaMfs.c [hunk 1]
```diff
@@ -381,6 +399,15 @@ Gia_Man_t * Gia_ManInsertMfs( Gia_Man_t * p, Sfm_Ntk_t * pNtk, int fAllBoxes )
             continue;
         }
         Vec_IntClear( vLeaves );
+
+        // handle internal CIs first, before we go looking for vLeaves
+        if ( iGroup != -1 && Abc_LitIsCompl(iGroup) )
+        {
+            //Dau_DsdPrintFromTruth( pTruth, Vec_IntSize(vLeaves) );
```
Comment (Ravenslofty) [inline src/aig/gia/giaMfs.c:406]: maybe this line should be cleaned up.

**Review by Ravenslofty — APPROVED** [review]
Tiny nitpick, but thanks to @yrabbit [generously testing](https://mastodon.sdf.org/@yrabbit/112172418888475022) this on the Project Apicula test suite, I am reasonably confident this doesn't break code in the wild, and even helps in some cases.

File: src/aig/gia/giaMfs.c (hunk 1 above)
Comment (povik) [reply src/aig/gia/giaMfs.c:406]: In the ABC coding syle I am not sure this is a line waiting to be cleaned up, rather a line waiting to be uncommented for debugging. Lines like these are everywhere so I don't bother, to be frank

Comment (Ravenslofty) [reply src/aig/gia/giaMfs.c:406]: Yeah, that's fine.

**Comment by QuantamHD** [comment]
Should this be upstreamed to abc?

**Comment by povik** [comment]
@QuantamHD Yes, but because of discouraging stories about how others tried to upstream ABC fixes, and because `&mfs -b` would need extra handling in this fix, I haven't opened a PR so far.

File: src/opt/sfm/sfmWin.c [hunk 2]
```diff
@@ -170,14 +171,33 @@ Vec_Int_t * Sfm_NtkDfs( Sfm_Ntk_t * p, Vec_Wec_t * vGroups, Vec_Int_t * vGroupMa
     Vec_IntClear( vBoxesLeft );
     vNodes = Vec_IntAlloc( p->nObjs );
     Sfm_NtkIncrementTravId( p );
-    if ( fAllBoxes )
+
+    assert(!fAllBoxes); // TODO
```
Comment (povik) [inline src/opt/sfm/sfmWin.c:175]: See here; this assertion fails for `&mfs -b`

**Comment by QuantamHD** [comment]
> @QuantamHD Yes, but because of discouraging stories about how others tried to upstream ABC fixes, and because `&mfs -b` would need extra handling in this fix, I haven't opened a PR so far.

We have a good relationship with Alan. I think it's worth opening a PR if you can, and I can signal boost it with him.

**Pull request merged**